lib/src/chrome.dart

```diff
@@ -1,3 +1,244 @@
+
+const _chromeEnvironment = 'CHROME_EXECUTABLE';
+const _linuxExecutable = 'google-chrome';
+const _macOSExecutable =
+    '/Applications/Google Chrome.app/Contents/MacOS/Google Chrome';
+const _windowsExecutable = r'Google\Chrome\Application\chrome.exe';
+var _windowsPrefixes = [
+  Platform.environment['LOCALAPPDATA'],
+  Platform.environment['PROGRAMFILES'],
+  Platform.environment['PROGRAMFILES(X86)']
+];
+
+String get _executable {
+  if (Platform.environment.containsKey(_chromeEnvironment)) {
+    return Platform.environment[_chromeEnvironment];
+  }
+  if (Platform.isLinux) return _linuxExecutable;
+  if (Platform.isMacOS) return _macOSExecutable;
+  if (Platform.isWindows) {
+    return p.join(
+        _windowsPrefixes.firstWhere((prefix) {
+          if (prefix == null) return false;
+          final path = p.join(prefix, _windowsExecutable);
+          return File(path).existsSync();
+        }, orElse: () => '.'),
+        _windowsExecutable);
+  }
+  throw StateError('Unexpected platform type.');
+}
+
+var _currentCompleter = Completer<Chrome>();
+
+/// A class for managing an instance of Chrome.
+class Chrome {
+  Chrome._(
+    this.debugPort,
+    this.chromeConnection, {
+    Process process,
+  }) : _process = process;
+
+  final int debugPort;
+  final Process _process;
+  final ChromeConnection chromeConnection;
+
+  /// Connects to an instance of Chrome with an open debug port.
+  static Future<Chrome> fromExisting(int port) async =>
+      _connect(Chrome._(port, ChromeConnection('localhost', port)));
+
+  static Future<Chrome> get connectedInstance => _currentCompleter.future;
+
+  /// Starts Chrome with the given arguments and a specific port.
+  ///
+  /// Each url in [urls] will be loaded in a separate tab.
+  static Future<Chrome> startWithPort(
+    List<String> urls, {
+    String userDataDir,
+    int remoteDebuggingPort,
+    bool disableBackgroundTimerThrottling = false,
+    bool disableExtensions = false,
+    bool disablePopupBlocking = false,
+    bool bwsi = false,
+    bool noFirstRun = false,
+    bool noDefaultBrowserCheck = false,
+    bool disableDefaultApps = false,
+    bool disableTranslate = false,
+  }) async {
+    final port = remoteDebuggingPort == null || remoteDebuggingPort == 0
+        ? await findUnusedPort()
+        : remoteDebuggingPort;
+
+    final process = await _startProcess(
+      urls,
+      userDataDir: userDataDir,
+      remoteDebuggingPort: port,
+      disableBackgroundTimerThrottling: disableBackgroundTimerThrottling,
+      disableExtensions: disableExtensions,
+      disablePopupBlocking: disablePopupBlocking,
+      bwsi: bwsi,
+      noFirstRun: noFirstRun,
+      noDefaultBrowserCheck: noDefaultBrowserCheck,
+      disableDefaultApps: disableDefaultApps,
+      disableTranslate: disableTranslate,
+    );
+
+    // Wait until the DevTools are listening before trying to connect.
+    await process.stderr
```

Comment:
Add a TODO to merge in https://github.com/dart-lang/webdev/pull/341 as this logic is problematic.